### PR TITLE
Feature: Labels and Auto New Channels (v2 API)

### PR DESCRIPTION
**Pull request by erichosick**
## Change description

Support for new channel "types" without having to update code: specifically for user category preferences.

This version of our react-headless will only work against v2 of the MagicBell backend.

* /notification_preferences now has a label property for each category.
* Consumes the v2 version of /notification_preferences.
* When a user saves preferences and a new channel has been added since the last save, that channel will now be available.
* Tests now deepFreeze objects to assure library is immutable.
* Improved test coverage

## Type of change
- [ ] Bug fix (fixes an issue)
- [X] New feature (adds functionality)

### Development

- [X] Lint rules pass locally
- [X] Application changes have been tested thoroughly
- [X] Automated tests covering modified code pass

### Security

- [X] Security impact of change has been considered
- [X] Code follows company security practices and guidelines

### Code review 

- [X] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: src/lib/ajax.ts [hunk 1]
```diff
@@ -28,6 +28,8 @@ export function buildAPIHeaders() {
   const headers = {
     'X-MAGICBELL-CLIENT-ID': clientId,
     'X-MAGICBELL-API-KEY': apiKey,
+    // From this point on, react headless only works against the v2 backend.
+    'Accept-Version': 'v2',
```
Comment (josuemontano) [inline src/lib/ajax.ts:32]: @erichosick Since we are releasing v2 slowly, could you please set this header only for the requests made to the user preferences endpoint?

**Review by josuemontano — CHANGES_REQUESTED** [review]

File: src/stores/notification_preferences/NotificationPreferencesRepository.ts [hunk 2]
```diff
@@ -23,21 +22,48 @@ export default class NotificationPreferencesRepository {
...
+    try {
+      const url = this.remotePathOrUrl;
+      const payload = humps.decamelizeKeys({ notificationPreferences: data });
+      const json = await putAPI(url, payload);
+
+      return humps.camelizeKeys(json);
+    } catch (error) {
+      // To maintain interface compatability as much as possible, we need to
+      // return a 403 error but false on a 500 error based on existing test.
+      if (error.message === 'Request failed with status code 403') {
```
Comment (josuemontano) [inline src/stores/notification_preferences/NotificationPreferencesRepository.ts:62]: @erichosick Can we check for the status code instead of the message, please?
